cdk-testsuite/tests/vagrant_sshfs.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python
+
+from avocado import Test
+from avocado.utils import process
+import os, re, vagrant, platform
+
+class VagrantSshfs(Test):
+    def setUp(self):
+        self.vagrant_VAGRANTFILE_DIR = self.params.get('vagrant_VAGRANTFILE_DIR')
+        self.vagrant_PROVIDER = self.params.get('vagrant_PROVIDER', default='')
+        self.vagrant_RHN_USERNAME = self.params.get('vagrant_RHN_USERNAME')
+        self.vagrant_RHN_PASSWORD = self.params.get('vagrant_RHN_PASSWORD')
+	self.sudo_PASSWORD = self.params.get('sudo_PASSWORD')
+	self.platform = platform.system()
+	if "CYGWIN" in self.platform:
+	    self.mountpoint = os.getenv("USERPROFILE")
+	    self.mountpoint_vm = '/' + self.mountpoint[:1].lower() + self.mountpoint[2:].replace("\\", "/")
+	    self.mountpoint_host = "/cygdrive" + self.mountpoint_vm
+        else:
+            self.mountpoint = self.mountpoint_host = self.mountpoint_vm = os.getenv("HOME")
+	self.dummy_file1_vm = os.path.join(self.mountpoint_vm, "dummy_file1.txt")
+	self.dummy_file2_vm = os.path.join(self.mountpoint_vm, "dummy_file2.txt")
+	self.dummy_file3_vm = os.path.join(self.mountpoint_vm, "dummy_file3.txt")
+	self.dummy_file1_host = os.path.join(self.mountpoint_host, "dummy_file1.txt")
+	self.dummy_file2_host = os.path.join(self.mountpoint_host, "dummy_file2.txt")
+	self.dummy_file3_host = os.path.join(self.mountpoint_host, "dummy_file3.txt")
+	self.dummy_contents1 = "Dumping dummy contents into file"
+	self.dummy_contents2 = "This is a dummy file"	
+	self.v = vagrant.Vagrant(self.vagrant_VAGRANTFILE_DIR)
+	os.chdir(self.vagrant_VAGRANTFILE_DIR)
+
+    
+    def vagrant_up_with_subscription(self):
+	''' vagrant up with registration to RHN '''
+	#self.vagrant_destroy()
+        self.log.info("Brining up the vagrant box and registering to RHN...")
+        os.environ["SUB_USERNAME"] = self.vagrant_RHN_USERNAME
```

Comment:
It would be good to be able to use this test even without registration.
I think solution can be the same as in vagrant_sanity.py, which mean that such function should probably be in library.